File: Demos/Datasnap/Server/proxy/java_android/com/embarcadero/javaandroid/TDBXStreamValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javaandroid;


/**
 * 
 * Wraps the {@link TStream} type and allows it to be null
 *
 */

public class TDBXStreamValue extends DBXValue {
	protected boolean ValueNull = false;
	
	public TDBXStreamValue() {
		super();
		setDBXType(DBXDataTypes.BinaryBlobType);
	}
	
	public void setNull() {
		ValueNull = true;
		streamValue = null;
	}

	public boolean isNull() {
		return ValueNull;
	}
	
	@Override
	public void SetAsStream(TStream Value) {
		streamValue = Value;
	}

	@Override
	public TStream GetAsStream() throws DBXException {
		return streamValue;
	}

}

File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TDBXStreamValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXStreamValue.h"


@implementation TDBXStreamValue
-(id) init {
	self = [super init];
	if (self) {
		[self setDBXType:BinaryBlobType];
		ValueNull =  NO;
	}
	return self;
}
-(void) SetNull {
	ValueNull = YES;
	streamValue = nil;
}

-(bool) isNull {
	return ValueNull;
}
-(NSData*) GetAsStream{
	[self checkCurrentDBXType:BinaryBlobType];
	return streamValue;
}
-(void) SetAsStream:(NSData *) value{
	[streamValue release];
	streamValue = [value retain];
	[self setDBXType:BinaryBlobType];
}




@end
